fms/models/__init__.py

```diff
@@ -19,28 +19,40 @@
     UniformModelParallelStrategy,
 )
 from fms.utils import serialization
+from fms.utils.config import ModelConfig
 
 
-__models: MutableMapping[str, MutableMapping[str, Callable[[], nn.Module]]] = {}
+__models: MutableMapping[
+    str, MutableMapping[str, Tuple[Callable[[], nn.Module], Optional[ModelConfig]]]
+] = {}
 
 
-def register_model(architecture: str, variant: str, factory: Callable[[], nn.Module]):
+def register_model(
+    architecture: str,
+    variant: str,
+    factory: Callable[[], nn.Module],
+    config: Optional[ModelConfig] = None,
+):
     """
-    Registers a model variant to be made available in the registration API.
+    Registers a model variant and its config to be made available in the registration API.
```

Comment:
might want to make it clearer that the config is only required for a particular use-case and what the use-case is